pulsar-client-cpp/lib/Consumer.cc

```diff
@@ -304,10 +305,19 @@ void Consumer::redeliverUnacknowledgedMessages() {
     }
 }
 
-Result Consumer::getConsumerStats(BrokerConsumerStats& BrokerConsumerStats, int partitionIndex) {
+Result Consumer::getConsumerStats(BrokerConsumerStats& brokerConsumerStats, int partitionIndex) {
     if (!impl_) {
         return ResultConsumerNotInitialized;
     }
-    return impl_->getConsumerStats(BrokerConsumerStats, partitionIndex);
+    Promise<Result, BrokerConsumerStats> promise;
+    impl_->getConsumerStatsAsync(WaitForCallbackValue<BrokerConsumerStats>(promise), partitionIndex);
```

Comment:
I can call Consumer-> getConsumerStatsAsync() instead of impl_-> getConsumerStatsAsync
but I saw the code for acknowledge() and unsubscribe() - they both call the async function in the impl_